react/utils/normalize.ts

```diff
@@ -67,6 +76,96 @@ export function changeImageUrlSize(
   return `${normalizedImageUrl}${queryStringSeparator}width=${width}&height=${height}&aspect=true`
 }
 
+function getPriceBySeller(seller: { commertialOffer: { Price: number } }) {
+  return seller.commertialOffer.Price
+}
+
+/**
+ * @description
+ * Find the lowest/highest `Price` between sellers
+ * @param item an SKU
+ * @param condition the condition enum used for comparison functions
+ * */
+function getPriceByCondition(item: { sellers: any }, condition: ConditionRule) {
+  // First, if there's only 1 seller, avoid filtering
+  const { sellers } = item
+
+  if (sellers.length === 1) return sellers[0].commertialOffer.Price
+
+  const availableSellers = sellers.filter(
+    ({ commertialOffer }: any) =>
+      commertialOffer.AvailableQuantity != null &&
+      commertialOffer.AvailableQuantity > 0
+  )
+
+  let itemPrice
+
+  if (condition === 'expensive') {
+    itemPrice = Math.max(...availableSellers.map(getPriceBySeller))
+  } else {
+    itemPrice = Math.min(...availableSellers.map(getPriceBySeller))
+  }
+
+  return itemPrice
+}
+
+/**
+ * @description
+ * Retrieves the SKU with the best price considering a provided condition
+ * @param items an array of SKUs
+ * @param condition the condition enum used for comparison functions
+ * */
+function getBestSKUPrice(items: any[], condition: ConditionRule) {
+  // First, if none or only 1 sku is available, avoid reducing
+  const filteredItems = items.filter(getOnlyAvailable)
+  if (filteredItems.length === 0) return items[0]
+  if (filteredItems.length === 1) return filteredItems[0]
+
+  return filteredItems.reduce((acc: any, curr: any) => {
+    if (condition === 'expensive') {
+      return getPriceByCondition(curr, condition) >
+        getPriceByCondition(acc, condition)
+        ? curr
+        : acc
+    }
+
+    return getPriceByCondition(curr, condition) <
+      getPriceByCondition(acc, condition)
+      ? curr
+      : acc
+  })
```

Comment:
I think you could simplify this by doing something like:

```suggestion
    const price = getPriceByCondition(curr, condition)
    const currentBestPrice = getPriceByCondition(acc, condition)
   
    if (condition === 'expensive') {
      return price > currentBestPrice ? curr : acc
    }

    return price < currentBestPrice ? curr : acc
  })
```

Reply:
Added